src/init/features/functions/index.ts

```diff
@@ -30,6 +22,104 @@ export async function doSetup(setup: any, config: any, options: Options) {
       ensure(projectId, "runtimeconfig.googleapis.com", "unused", true),
     ]);
   }
+  setup.functions = {};
+
+  if (!config.src.functions) {
+    // if functions have not been initialized yet
+    setup.config.functions = [];
+    return codebaseSetup(setup, config, false);
+  } else {
+    // if functions have already been initialized
+    // make sure config is validated and normalized
+    setup.config.functions = normalizeAndValidate(setup.config.functions);
+
+    logger.info();
+    logger.info("Detected " + clc.bold("existing codebase(s)."));
+    logger.info();
+
+    const choices = [
+      {
+        name: "Initialize",
+        value: "new",
+      },
+      {
+        name: "Re-initialize",
+        value: "reinit",
+      },
+    ];
+    const initOpt = await promptOnce({
+      type: "list",
+      message: "Would you like to initialize a new codebase, or reinitialize an existing one?",
+      default: "new",
+      choices,
+    });
+    return codebaseSetup(setup, config, initOpt === "reinit");
+  }
+}
+
+/**
+ *  User dialogue to set up configuration for functions codebase.
+ */
+async function codebaseSetup(setup: any, config: Config, reinit: boolean): Promise<any> {
+  if (reinit) {
+    const choices = setup.config.functions.map((cbconfig: any) => {
+      return {
+        name: cbconfig["codebase"],
+        value: cbconfig["codebase"],
+      };
+    });
+    const codebase = await promptOnce({
+      type: "list",
+      message: "Which codebase would you like to re-initialize?",
+      choices,
+    });
+    const cbconfig = configForCodebase(setup.config.functions, codebase);
+    _.set(setup, "functions.source", cbconfig.source);
+    _.set(setup, "functions.codebase", cbconfig.codebase);
+
+    logger.info("Re-initializing " + clc.bold(`codebase ${codebase}...`));
+    logger.info();
+  } else {
+    logger.info();
+    logger.info("Let's create a new codebase to contain and manage your functions.");
+    logger.info("A directory corresponding to the codebase will be created in your project");
+    logger.info("with sample code pre-configured.");
```

Comment:
similar concern here - let's use a single `logger.info()` for each line rather than rely on that it won't print anything.